frontend/src/features/dsComponents/readme/Readme.tsx

```diff
@@ -11,26 +11,33 @@ export const ReadmeComponent: React.FunctionComponent<ReadmeProps> = (props) =>
   const { qriRef } = props
   const [hasReadme, setHasReadme] = React.useState(true)
 
-  const qriRefStr = refStringFromQriRef(qriRef)
-
   const refCallback = useCallback((el: HTMLDivElement) =>{
     if (el !== null) {
       fetch(`${API_BASE_URL}/dataset_summary/readme?path=${qriRef.path}`)
-        .then(async (res) => {
-          return res.text()
+        .then((res) => {
+          if (res.ok) {
+            return res.text()
+          } else {
+            return Promise.reject('error 404')
+          }
         })
-        .then((render) => {
-          if (!render) { setHasReadme(false) }
-          el.innerHTML = render
+        .then((renderedReadme) => {
+          el.innerHTML = renderedReadme
+        })
+        .catch((e) => {
+          setHasReadme(false)
         })
```

Comment:
In the future, use the `async / await` API instead of explicit promises whenever possible. This code is ripe for refactoring, so I don't think it needs a change this go-round